sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/filters/recurrence/RecurrenceEvaluatorTest.java

```diff
@@ -0,0 +1,534 @@
+        settings.setRecurrence(recurrence);
+        consumeEvaluationTestData(settings, now, false);
+    }
+
+    @Test
+    public void weeklyDaysOfWeekIntervalNumberedRangeTrue() {
+        final ZonedDateTime now4 = ZonedDateTime.parse("2023-09-17T00:00:00+08:00");    // Sunday in the 3rd week
+        final TimeWindowFilterSettings settings4 = new TimeWindowFilterSettings();
+        final RecurrencePattern pattern4 = new RecurrencePattern();
+        final RecurrenceRange range4 = new RecurrenceRange();
+        final Recurrence recurrence4 = new Recurrence();
+        pattern4.setType("Weekly");
+        pattern4.setDaysOfWeek(List.of("Monday", "Sunday"));
+        pattern4.setFirstDayOfWeek("Monday");
+        pattern4.setInterval(2);
+        range4.setType("Numbered");
+        range4.setNumberOfRecurrences(3);
+        recurrence4.setRange(range4);
+        recurrence4.setPattern(pattern4);
+        settings4.setStart("2023-09-03T00:00:00+08:00");    // Sunday
+        settings4.setEnd("2023-09-03T00:00:01+08:00");
+        settings4.setRecurrence(recurrence4);
+        consumeEvaluationTestData(settings4, now4, true);
+    }
+
+    @Test
+    public void weeklyTimeWindowAcrossDaysIntervalTrue() {
+        final ZonedDateTime now = ZonedDateTime.parse("2023-09-13T08:00:00+08:00");    // Within the recurring time window 2023-09-11T:00:00:00+08:00 ~ 2023-09-15T:00:00:00+08:00
+        final TimeWindowFilterSettings settings = new TimeWindowFilterSettings();
+        final RecurrencePattern pattern = new RecurrencePattern();
+        final RecurrenceRange range = new RecurrenceRange();
+        final Recurrence recurrence = new Recurrence();
+        pattern.setType("Weekly");
+        pattern.setDaysOfWeek(List.of("Sunday", "Monday")); // Time window occurrences: 9-3 ~ 9-7 (1st week), 9-11 ~ 9-15 and 9-17 ~ 9-21 (3rd week)
+        pattern.setFirstDayOfWeek("Monday");
+        pattern.setInterval(2);
+        recurrence.setRange(range);
+        recurrence.setPattern(pattern);
+        settings.setStart("2023-09-03T00:00:00+08:00");
+        settings.setEnd("2023-09-07T00:00:01+08:00");
+        settings.setRecurrence(recurrence);
+        consumeEvaluationTestData(settings, now, true);
+    }
+
+    @Test
+    public void weeklyTimeWindowAcrossDaysNumberedRangeTrue() {
+        final ZonedDateTime now5 = ZonedDateTime.parse("2023-09-19T00:00:00+08:00");    // The 3rd occurrence: 2023-9-17T:00:00:00+08:00 ~ 2023-9-21T:00:00:00+08:00
+        final TimeWindowFilterSettings settings5 = new TimeWindowFilterSettings();
+        final RecurrencePattern pattern5 = new RecurrencePattern();
+        final RecurrenceRange range5 = new RecurrenceRange();
+        final Recurrence recurrence5 = new Recurrence();
+        pattern5.setType("Weekly");
+        pattern5.setDaysOfWeek(List.of("Monday", "Sunday"));    // Time window occurrences: 9-3 ~ 9-7 (1st week), 9-11 ~ 9-15 and 9-17 ~ 9-21 (3rd week)
```

Comment:
The comment in this line is not correct. The `FirstDayOfWeek` is set to Monday. The 1st week should be 2023.9.3 (just only one day, the Sunday is the last day of the 1st week) and the 2nd week should be 2023.9.11~ 2023.9.17.

Reply:
Oh, I found this is copied from .NET FM. I will update it.